Tutorials/Style/Functions-Docstrings.MD

```diff
@@ -0,0 +1,169 @@
+
+```python
+def square(x):
+    '''This is my docstring'''
+    return x**2
+```
+
+Now that we have our docstring, it will show up when we run `help()` on our function! This is a major advantage of using docstrings instead of comments when documenting functions. For instance: 
+
+```python
+def square(x):
+    '''This is my docstring'''
+    return x**2
+
+help(square)
+```
+
+returns
+
+```
+Help on function square in module __main__:
+
+square(x)
+    This is my docstring
+
+```
+
+which tells us that the function `square` is defined in the main code, it takes one input `x`, and that... it has a docstring that says `this is my docstring`. That's a docstring alright, but it isn't a very helpful one...
+
+### How do I write a _good_ docstring?
+
+It is only slightly harder to write a good docstring than it is to write a docstring. Best practices for writing a docstrings are: 
+- Describe what the function _does_ (this is usually the first line)
+- Describe what the function takes as an input, including data types, formats, variable names, and units if applicable
+- Describe what the function gives as an output, includeing data types, formats, and units if applicable
+- Note any domains over which the function will/will not work (for instance: does this work for integers only? Positive numbers?)
+- Note any dependencies the function has on other packages, functions, object types, etc. 
+Note that with the triple quotes, you can write a docstring that spans multiple lines, so if this seems like a lot to include, you have as much room as you need. 
+
+Here is how I would document the `square` function:
+
+```python
+def square(x):
+    '''
+    Squares the input. 
+    
+    The square function takes a single input:
+    - x: an integer or float
+    and returns: 
+    - x**2: an integer or float that is the square of the input x
+    '''
+    return x**2
+```
+
+Now when we run `help(square)` we get: 
+
+```
+Help on function square in module __main__:
+
+square(x)
+    Squares the input.
+    
+    The square function takes a single input:
+    - x: an integer or float
+    and returns: 
+    - x**2: an integer or float that is the square of the input x
+
+```
+
+Much better! Of course, this is a simple example that you could probably figure out just from the function definition, but docstrings are very useful when writing more complicated functions with multiple inputs, complicated outputs, or functions that require input of a certain format, unit, or data type. They help future readers (including yourself!) understand the code you've written. 
+
+Docstrings should also be used when defining python classes--recall how we can also use `help()` on objects like lists. For classes, the docstring goes in the same place as for functions: in triple quotes, beneath the class definition and above the class methods.
```

Comment:
Maybe add a reference for further reading, or more info about class-specific docstrings?